spec/msfl/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+shared_examples_for "an invocation of MSFL::Sinatra.validate" do
+
+  let(:params) { { dataset: dataset, filter: filter } }
```

Comment:
define let(:filter) and let(:dataset) here with argument errors, i like that pattern so its clear when a test doesn't have them defined and they are needed.